apps/andi/test/unit/andi_web/controllers/api/ingestion_controller_test.exs

```diff
@@ -162,13 +172,57 @@ defmodule AndiWeb.API.IngestionControllerTest do
 
   describe "PUT /ingestion" do
     test "PUT /api/ with data returns a 201", %{conn: conn} do
+      smrt_ingestion = TDG.create_ingestion(%{})
+      {_, ingestion_without_id} = smrt_ingestion |> struct_to_map_with_string_keys() |> Map.pop("id")
+
+      allow(Brook.Event.send(@instance_name, any(), any(), any()), return: :ok)
+      allow(Andi.InputSchemas.Datasets.get(any()), return: %{technical: %{sourceType: "ingest"}})
+
+      conn = put(conn, @route, ingestion_without_id)
+
+      {_, decoded_body} = Jason.decode(response(conn, 201))
+
+      expected_ingestion = TDG.create_ingestion(Map.put(ingestion_without_id, "id", decoded_body["id"]))
+
+      assert_called(Brook.Event.send(@instance_name, ingestion_update(), :andi, expected_ingestion))
+
+      assert_called(
+        Andi.Schemas.AuditEvents.log_audit_event(:api, ingestion_update(), expected_ingestion),
+        once()
+      )
+    end
+
+    test "PUT /api/ creating a ingestion with a set id returns a 400", %{conn: conn} do
       smrt_ingestion = TDG.create_ingestion(%{})
       ingestion = smrt_ingestion |> struct_to_map_with_string_keys()
+
       allow(Brook.Event.send(@instance_name, any(), any(), any()), return: :ok)
       allow(Andi.InputSchemas.Datasets.get(any()), return: %{technical: %{sourceType: "ingest"}})
-      put(conn, @route, ingestion)
+      allow(IngestionStore.get(any()), return: {:ok, nil})
```

Comment:
I like the use of expect here to confirm it was called. Preference though.

Regardless, locking down the ID as the arg is probably a good idea

expect(Ingestions.get(smart_ingestion.id), return: {:ok, nil})